### DISCUSSION TITLE
Installation binaries not being included in repo

### DISCUSSION THREAD

**Discussion by DiskCrasher** [JuliaLang/julia#55573] [General]
Is there a valid reason why the Windows binary installers aren't included in the GitHub repo along with the source code? Instead, one has to download them separately from https://julialang.org/downloads/. Our company mirrors GitHub repos internally and w/o the binary installer being included this doesn't help because we're not allowed to download from the Internet. Many other GH repos I've used include the binary installers in their releases which to me makes sense, having everything in one location.

**Comment by DilumAluthge** [comment]
Just to clarify, do you mean uploading the Julia release binaries to the "Assets" section of the GitHub Releases?

Or do you mean checking the release binary files into source control, that is, into the Git repo itself?

**Reply by KristofferC** [reply]
99.9% the asset section on a release. The other doesn't make sense.

**Reply by DiskCrasher** [reply]
Under the Assets section with the rest of the compressed binary files.

**Reply by DilumAluthge** [reply]
So, just to clarify, when you say that your company "mirrors GitHub repos internally", what specifically are they mirroring?

Generally, when someone says that they are mirroring a Git repo, I usually interpret that to mean that they are only mirroring the Git repo itself. (Which is why I asked you the above clarification question.)

---

A few other thoughts:

1. There's no particularly reason that we don't currently upload the official release binaries to the "Assets" section of GitHub Releases section. I suppose that we could start doing that.
    - One potential disadvantage is that we won't get the same level of statistics as far as how many users are downloading Julia, when they are downloading it, etc. (The GitHub API offers some download statistics, but AFAICT not to the same level of granularity that we currently get by hosting the release binaries on the JuliaLang.org website.)
    - Also, this probably has to be done manually for now, so it's kind of annoying to add an additional manual step for the release managers. But that's not necessarily a sufficient reason by itself to not do it.
2. Note that if your company blocks outgoing requests to the Internet, then you almost certainly won't be able to install Julia packages unless your company adds firewall exceptions for the [relevant JuliaLang domains](https://julialang.org/downloads/#official_domains) (and at that point you could also just download Julia from the JuliaLang website). So you would be able to use Base Julia itself, but you wouldn't be able to install any Julia packages, which for most people would be severely limiting.

**Reply by DiskCrasher** [reply]
I'm not the one who sets up the mirrors/proxies but in looking at some other projects it appears they mirror the (typically binary) files found under the Releases section of the repos. They may also mirror the source code found in the main repo itself, depending on the project. In this case, I'm specifically interested in the released binaries. Hopefully that answers your question.

As for external Julia packages, they've set up their own internal Julia package server. And yes, there are definitely environments where Internet access won't be available. (Even when it is it's still against policy to download from external sites.)

**Reply by vchuravy** [reply]
As a counterpoint, many large projects that don't exclusively use GitHub infrastructure, don't publish assets under release tags. As an example, take Python, they only have binaries available from their website. So for us, the question is whether the extra work and the duplication is worth it. I am not convinced.

**Reply by DilumAluthge** [reply]
> So for us, the question is whether the extra work and the duplication is worth it. I am not convinced.

Yeah, I'm a little concerned about the extra burden (and workload) that this places on the release managers.

**Reply by DilumAluthge** [reply]
> As for external Julia packages, they've set up their own internal Julia package server.

I wonder if the easiest approach is to just have the package server also serve the Julia release binaries.

**Reply by DiskCrasher** [reply]
The binaries are fetched separately through an approved process. Problem is, someone has to submit a request to have them updated to the latest version (not automated) which is why I suggested adding them to the repo (automated via mirror/proxy).
